cobra/flux_analysis/deletion.py

```diff
@@ -0,0 +1,348 @@
+        to be deleted.
+
+    method: {"fba", "moma"}, optional
+        Procedure used to predict the growth rate.
+
+    solver: str, optional
+        This must be a QP-capable solver for MOMA. If left unspecified,
+        a suitable solver will be automatically chosen.
+
+    zero_cutoff: float, optional
+        When checking to see if a value is 0, this threshold is used.
+
+    Returns
+    -------
+    dict of dict
+        A sparse representation of all combinations of pairs of reaction
+        deletions without replacements.
+    """
+    with cobra_model as model:
+        WORKERS = dict(gene=_gene_deletion_worker, reaction=_reaction_deletion_worker)
+        worker_function = WORKERS[entity]
+
+        try:
+            (legacy, solver) = sutil.choose_solver(model, solver, qp=(method == "moma"))
+        except sutil.SolverNotFound:
+            if method == "moma":
+                warn("Cannot use MOMA since no QP-capable solver was found. Falling"
+                     " back to FBA.")
+                (legacy, solver) = sutil.choose_solver(model, solver)
+            else:
+                (err_type, err_val, err_tb) = sys.exc_info()
+                raise_(err_type, err_val, err_tb)  # reraise for Python2&3
+
+        if legacy:
+            raise ValueError(
+                "Legacy solvers are not supported any longer. Please use one of"
+                " the optlang solver interfaces instead.")
+
+        if num_cpu is None:
+            num_cpu = kwargs.get("number_of_processes")
+        if num_cpu is None:
+            try:
+                num_cpu = multiprocessing.cpu_count()
+            except NotImplementedError:
+                warn("Number of cores could not be detected - assuming 1.")
+                num_cpu = 1
+
+        # Reactions with 0 flux in the current conditions have no impact on
+        # deletion. Determine those reactions now to speed up computation later.
+        if method == 'moma':
+            moma.add_moma(model)
+        if method == 'linear moma':
+            moma.add_moma(model, linear=True)
+        solution = model.optimize()
+        if solution.status == "optimal":
+            kwargs["growth"] = solution.objective_value
+            kwargs["zero_flux_reactions"] = \
+                {rxn for (rxn, flux) in solution.fluxes.items()
+                 if abs(flux) < zero_cutoff}
+        else:
+            warn("Non-optimal ({0:s}) base solution.".format(solution.status))
+
+        def unordered_combination(reactions):
+            return frozenset([i.id for i in reactions if i.id not in kwargs["zero_flux_reactions"]])
+
+        args = set([unordered_combination(comb) for comb in product(*element_lists)])
+        worker_function.model = model
+        if num_cpu > 1:
+            chunk_size = len(args) // num_cpu
+            pool = multiprocessing.Pool(num_cpu)
+            results = pool.imap_unordered(worker_function, args, chunksize=chunk_size)
```

Comment:
I think the pool has to be terminated or closed after running the map. Otherwise you will spawn a lot of zombie processes.